### PR TITLE
[IGNORE] Add comments + stylistic updates for number formatting code

### PR DESCRIPTION
**Pull request by cndonovan**
Stylistic changes to the number formatting code -- No functional changes

Hopefully these changes make it easier for anyone new to this code to figure out what's going on.

* Added comments
* Alphabetized code
* Made code consistent 
* Made it more obvious what the formatter is

### CODE REVIEW

File: ui/core/src/model/units/bytes.ts [hunk 1]
```diff
@@ -32,17 +40,14 @@ export const BYTES_GROUP_CONFIG: UnitGroupConfig = {
   abbreviate: true,
 };
 export const BYTES_UNIT_CONFIG: Readonly<Record<BytesUnitKind, UnitConfig>> = {
-  // These units are powers of 1000.
-  // In other words, 1 KB = 1000 bytes.
```
Comment (cndonovan) [inline ui/core/src/model/units/bytes.ts:36]: Moved this to the top of the file -- make people generally aware of our conventions at the beginning of this file

File: ui/core/src/model/units/bytes.ts [hunk 2]
```diff
@@ -32,17 +40,14 @@ export const BYTES_GROUP_CONFIG: UnitGroupConfig = {
...
   Bytes: {
     group: 'Bytes',
     label: 'Bytes',
   },
 };
 
-export function formatBytes(bytes: number, options: BytesUnitOptions) {
-  const { abbreviate, decimal_places } = options;
-
+export function formatBytes(bytes: number, { abbreviate, decimal_places }: BytesUnitOptions) {
```
Comment (cndonovan) [inline ui/core/src/model/units/bytes.ts:49]: Destructuring parameters inside the arguments themselves -- for consistency with other files

File: ui/core/src/model/units/units.ts [hunk 3]
```diff
@@ -11,12 +11,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { BytesUnitOptions, BYTES_GROUP_CONFIG, BYTES_UNIT_CONFIG, formatBytes } from './bytes';
-import { DecimalUnitOptions, DECIMAL_UNIT_CONFIG, formatDecimal, DECIMAL_GROUP_CONFIG } from './decimal';
-import { PERCENT_GROUP_CONFIG, formatPercent, PercentUnitOptions, PERCENT_UNIT_CONFIG } from './percent';
+import { formatBytes, BytesUnitOptions, BYTES_GROUP_CONFIG, BYTES_UNIT_CONFIG } from './bytes';
+import { formatDecimal, DecimalUnitOptions, DECIMAL_GROUP_CONFIG, DECIMAL_UNIT_CONFIG } from './decimal';
+import { formatPercent, PercentUnitOptions, PERCENT_GROUP_CONFIG, PERCENT_UNIT_CONFIG } from './percent';
```
Comment (cndonovan) [inline ui/core/src/model/units/units.ts:16]: Alphabetized so it's easy to see at a glance

File: ui/core/src/model/units/units.ts [hunk 4]
```diff
@@ -11,12 +11,20 @@
...
 import { formatTime, TimeUnitOptions, TIME_GROUP_CONFIG, TIME_UNIT_CONFIG } from './time';
 import { UnitGroup, UnitGroupConfig, UnitConfig } from './types';
 
+/**
+ * Most of the number formatting is based on Intl.NumberFormat, which is built into JavaScript.
+ * Prefer Intl.NumbeFormat because it covers most use cases and will continue to be well-supported with time.
+ *
+ * To format bytes, we also make use of the `numbro` package,
+ * because it can handle adding units like KB, MB, GB, etc. correctly.
+ */
```
Comment (cndonovan) [inline ui/core/src/model/units/units.ts:26]: Explain package usage at the top of the parent file

File: ui/core/src/model/units/units.ts [hunk 5]
```diff
@@ -40,20 +48,20 @@ export function formatValue(value: number, unitOptions?: UnitOptions): string {
     return value.toString();
   }
 
-  if (isDecimalUnit(unitOptions)) {
-    return formatDecimal(value, unitOptions);
+  if (isBytesUnit(unitOptions)) {
+    return formatBytes(value, unitOptions);
```
Comment (cndonovan) [inline ui/core/src/model/units/units.ts:52]: Alphabetized because why not

File: ui/core/src/model/units/bytes.ts [hunk 6]
```diff
@@ -60,11 +65,12 @@ export function formatBytes(bytes: number, options: BytesUnitOptions) {
         formatterOptions.maximumSignificantDigits = MAX_SIGNIFICANT_DIGITS;
       }
     }
-
-    return Intl.NumberFormat('en-US', formatterOptions).format(bytes);
+    const formatter = Intl.NumberFormat('en-US', formatterOptions);
+    return formatter.format(bytes);
```
Comment (cndonovan) [inline ui/core/src/model/units/bytes.ts:69]: More verbose but probably faster to grok things when you're new to the file

**Review by sjcobb — CHANGES_REQUESTED** [review]
🙏

**Review by sjcobb — APPROVED** [review]
Whoops meant to approve!

**Pull request merged**